pkgs/tools/text/ripgrep/default.nix

```diff
@@ -7,10 +7,12 @@
 , Security
 , withPCRE2 ? true
 , pcre2
-, enableManpages ? stdenv.hostPlatform.emulatorAvailable buildPackages
 }:
 
-rustPlatform.buildRustPackage rec {
+let
+  canRunRg = stdenv.hostPlatform.emulatorAvailable buildPackages;
+  rg = "${stdenv.hostPlatform.emulator buildPackages} $out/bin/rg";
```

Comment:
Why do we always need to use an emulator? Shouldn't we avoid those and rather use buildPackages.rg?

Reply:
my reasoning is that it's not easy to be sure that `rg`'s generated manual (and shell completions, etc) will be the same for all targets. presumably upstream now generates the manual/completions instead of shipping it in the source like they used to _because_ it depends on compile-time configuration (e.g. whether or not it's built with PCRE2) -- which could differ between this derivation and the `buildPackages.ripgrep` derivation.

Reply:
> Shouldn't we avoid those

Aren't emulators for precisely that case, the program itself is required for some assets (such as documentation), but you can't invoke it directly because you're doing a cross-build?

Is there any previous discussion btw? Otherwise I'm inclined to merge and an issue should be opened instead where the use of buildPackages.X vs emulators should be discussed so that we have a consistent decision and don't need to negotiate that on every PR that makes use of that.

Reply:
fine for me